dns/dns_updater.go

```diff
@@ -167,18 +185,39 @@ func (u *updater) Update(entries controller.IngressEntries) error {
 	return nil
 }
 
-func (u *updater) determineManagedRecordSets(rrs []*route53.ResourceRecordSet) []*route53.ResourceRecordSet {
+func (u *updater) flattenRoute53ResourceRecordSet(rrs []*route53.ResourceRecordSet) []record {
+	var records []record
+
+	for _, recordSet := range rrs {
+		if len(recordSet.ResourceRecords) == 1 {
+			records = append(records, record{
+				name:     *recordSet.Name,
+				pointsTo: *recordSet.ResourceRecords[0].Value,
+			})
```

Comment:
How come we special case `len(recordSet.ResourceRecords) == 1`? Won't this break in arbitrary scenarios?

Reply:
yes, you're right.

It should be explicit and look for the types ALIAS and CNAME